src/Patches.php

```diff
@@ -176,7 +176,7 @@ public function gatherPatches(PackageEvent $event) {
     $operations = $event->getOperations();
     $this->io->write('<info>Gathering patches for dependencies. This might take a minute.</info>');
     foreach ($operations as $operation) {
-      if ($operation->getJobType() == 'install' || $operation->getJobType() == 'update') {
+      if ($operation instanceof InstallOperation || $operation instanceof UpdateOperation) {
```

Comment:
This line should be fixed, so it will work for composer v1 and v2

Reply:
Fixed how? I tested this (changed) line on both composer 1 and 2 and it works fine.

Reply:
Ok cool, didnt expect that. Even better.

I will also test this patch tomorrow, in our pipeline <3

Reply:
Just to be sure, are we sure that it works with _both_ versions? (Without test coverage it is hard to tell ;S)
I remember that I created a different fix when I experimented with Composer 2x on my project.
https://github.com/cweagans/composer-patches/compare/1.x...mxr576:1.x#diff-cf7534e34ca749ac0a88fd7f5dbe0869

Reply:
Test coverage is desirable but I don't think we should block this change waiting for it.

At first I also wrote a solution using `\Composer\Composer::RUNTIME_API_VERSION` but this solution is cleaner.

Reply:
I added initial setup for github actions with https://github.com/cweagans/composer-patches/pull/331